### PR TITLE
Change CustomerBillingData to resource and create via factory

### PR DESCRIPTION
**Pull request by arti0090**

### CODE REVIEW

**Review by SirDomin — APPROVED** [review]

File: src/Entity/CustomerBillingData.php [hunk 1]
```diff
@@ -46,80 +46,108 @@ class CustomerBillingData implements CustomerBillingDataInterface
...
+    {
+        $this->firstName = $firstName;
+    }
+
+    public function getLastName(): string
     {
         return $this->lastName;
     }
 
-    public function fullName(): string
```
Comment (Tomanhez) [inline src/Entity/CustomerBillingData.php:86]: We should add note in upgrade file about changed `public function fullName(): string` to `public function getStreet(): string` and all others

File: src/Entity/CustomerBillingData.php [hunk 2]
```diff
@@ -46,80 +46,108 @@ class CustomerBillingData implements CustomerBillingDataInterface
...
-        $this->firstName = $firstName;
-        $this->lastName = $lastName;
-        $this->street = $street;
-        $this->postcode = $postcode;
-        $this->countryCode = $countryCode;
-        $this->city = $city;
-        $this->company = $company;
-        $this->provinceName = $provinceName;
-        $this->provinceCode = $provinceCode;
+    public function getId(): ?int
```
Comment (GSadee) [inline src/Entity/CustomerBillingData.php:49]: I would be for not forcing the type of identifier
```suggestion
    public function getId()
```

File: src/Entity/CustomerBillingData.php [hunk 3]
```diff
@@ -46,80 +46,108 @@ class CustomerBillingData implements CustomerBillingDataInterface
...
+    {
+        return $this->id;
     }
 
-    public function id(): ?int
+    public function setId(?int $id): void
```
Comment (GSadee) [inline src/Entity/CustomerBillingData.php:54]:
```suggestion
    public function setId($id): void
```

File: src/Entity/CustomerBillingDataInterface.php [hunk 4]
```diff
@@ -13,27 +13,47 @@
 
 namespace Sylius\RefundPlugin\Entity;
 
-interface CustomerBillingDataInterface
+use Sylius\Component\Resource\Model\ResourceInterface;
+
+interface CustomerBillingDataInterface extends ResourceInterface
 {
-    public function id(): ?int;
+    public function setId(?int $id): void;
```
Comment (GSadee) [inline src/Entity/CustomerBillingDataInterface.php:20]:
```suggestion
    public function setId($id): void;
```

File: UPGRADE.md [hunk 5]
```diff
@@ -23,6 +23,48 @@
...
+    -   public function postcode(): string
+    +   public function getPostcode(): string
+    -   public function city(): string
+    +   public function getCity(): string
+    -   public function company(): ?string
+    +   public function getCompany(): ?string
+    -   public function provinceName(): ?string
+    +   public function getProvinceName(): ?string
+    -   public function provinceCode(): ?string
+    +   public function getProvinceCode(): ?string
```
Comment (GSadee) [inline UPGRADE.md:65]: I'm not sure if we should change these getters, probably the better way would be not changing them or adding new ones and leaving old methods with deprecation

Comment (arti0090) [reply UPGRADE.md]: I have added deprecated annotations and leaved a note in file about upcoming changes 🙂

File: UPGRADE.md [hunk 6]
```diff
@@ -23,6 +23,56 @@
...
+        }
+    ```
+
+1. The `id` method from `Sylius\RefundPlugin\Entity\CustomerBillingData` has its return type removed:
+    
+    ```diff
+    -   public function id(): ?int;
+    +   public function id();
+    ```
+
```
Comment (GSadee) [inline UPGRADE.md:49]:
```suggestion
```

**Review by GSadee — APPROVED** [review]

**Comment by GSadee** [comment]
Thank you, @arti0090! :1st_place_medal:

**Pull request merged**